src/clients/go/scripts/format_and_codegen.sh

```diff
@@ -1,10 +0,0 @@
-#!/usr/bin/env bash
```

Comment:
I think `fail_on_diff.sh` references this file

Reply:
I'm not familiar with what this script does exactly... may I remove the `fail_on_diff.sh` script too?

Reply:
:thinking: The script is supposed to run in CI, and fail if we forgot to auto-generate the bindings, but it looks like it never got added to CI.

Reply:
So, one less thing on the CI 🤝

Reply:
Actually I'm not sure? I know it will run automatically when CI builds & exports the golang client to the https://github.com/tigerbeetledb/tigerbeetle-go repo. But that doesn't guarantee that the bindings in the primary repo were updated (iiuc). Though I guess we could just not commit the go bindings (i.e. gitignore) — then they _can't_ get out of sync.

Reply:
We should do something like the C client does, checking the generated file against the current one.
Or, we can always generate and commit this file during the CI.

Reply:
I favor the former option because that is what we do elsewhere. I think this is what `fail_on_diff` was supposed to accomplish, it just wasn't hooked into CI.

Reply:
That's what the C client does: https://github.com/tigerbeetledb/tigerbeetle/blob/main/src/clients/c/tb_client_header_test.zig

I suggest we just generate the file again and compare byte-by-byte with the current one during CI.

Reply:
Ok, I liked the way the test is done, but it's now inconsistent with the `tb_client.h` test.
We generate the header for the C header test before testing, so the test will never fail during CI 🤔.

Reply:
Maybe in CI, after we run `zig build test`, we run `git diff` and check that there are no changes?

Reply:
Well, in this case, I think just generating the file with `zig build` and then running `git diff` would be enough to assert the current file is up to date. What would be the point of writing a test for that?

Reply:
That would be fine too. The "test" isn't important, what is important is that if the bindings need to be regenerated, that is caught by CI.

Reply:
Nice, so the current `go_binding_tests.zig` is good enough to catch it.

Reply:
Ahh, sorry, I got mixed up by

> We generate the header for the C header test before testing, so the test will never fail during CI

but you were referring to the C client. Yes, `go_binding_tests.zig` will catch it!

Reply:
Yes, I'd suggest removing [the C header dependency from the test](https://github.com/tigerbeetledb/tigerbeetle/blob/041ffbf06e4442071f928855676f61dea4764313/build.zig#L116), so it would be able to catch something wrong with the current header present on the repo.

But I think it's another PR.